src/main/io/serial.c

```diff
@@ -211,11 +212,13 @@ bool isSerialConfigValid(serialConfig_t *serialConfigToCheck)
                 return false;
             }
 
-            if (!(portConfig->functionMask & FUNCTION_MSP)) {
```

Comment:
You're right, but this is not related to the blackbox-serialRx sharing. 
This case (*FUNCTION_MSP | FUNCTION_RX_SERIAL*) is actually accepted but not tested as others are (like telemetry). Should add a *FUNCTION_RX_SERIAL*  in *ALL_FUNCTIONS_SHARABLE_WITH_MSP* (although tested in *ALL_FUNCTIONS_SHARABLE_WITH_BLACKBOX*)